karyon3-admin-simple/src/main/java/com/netflix/karyon/admin/CachingStaticResourceProvider.java

```diff
@@ -0,0 +1,35 @@
+package com.netflix.karyon.admin;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * Decorator for a StaticResourceProvider that caches the results
+ * 
+ * @author elandau
+ *
+ */
+public class CachingStaticResourceProvider implements StaticResourceProvider {
+
+    private final StaticResourceProvider delegate;
+    private final ConcurrentMap<String, StaticResource> templates = new ConcurrentHashMap<>();
+
+    public CachingStaticResourceProvider(StaticResourceProvider delegate) {
+        this.delegate = delegate;
+    }
+    
+    @Override
+    public CompletableFuture<StaticResource> getResource(String name) {
```

Comment:
Could/should this return a CompletableFuture<Optional<StaticResource>> ?  It would also avoid the INVALID constant below in StaticResource and the isValid check I think.